### DISCUSSION TITLE
ghelper model doesn't support custom fan curves

### DISCUSSION THREAD

**Discussion by zlqd123** [seerge/g-helper#3478] [General]
Today i try to update Ghelper 0.197 to 199. After i replaced Ghelper.exe and run it, i can't custom fan curves. Ghelper says "ghelper model doesn't support custom fan curves". 
So i come back to0.197, and it works  smoothly. I plan to write an issue to report this,so i rename Ghelper 199.exe and put it in the same folder of Ghelper 197.exe. Shutdown Ghelper 197.exe and run Ghelper 199.exe, then ver199 can work smoothly. 

Well, maybe it's just like reinstalling. Looks like i have fixed it.

[log - bug.txt](https://github.com/user-attachments/files/18122890/log.-.bug.txt)

ASUS TUF Gaming A16 FA607PI
Win11 23H2&24H2

**Comment by seerge** [comment]
@zlqd123 hello,

There are absolutely no difference in this regard between [0.199 and 0.197](https://github.com/seerge/g-helper/compare/v0.197...v0.199).

From your log I see that (custom) fan curve was set and accepted : 
```
12/13 星期五 15:16:25: App launched: ASUS TUF Gaming A16 FA607PI_FA607PI :0.199.0.0zh-CN.
...
12/13 星期五 15:16:27: FanCPU = 1E-38-3D-42-4B-52-58-61-00-01-02-0A-0A-10-21-5A : OK
12/13 星期五 15:16:27: FanGPU = 1E-38-3D-42-4C-52-58-61-00-01-07-0E-1F-1F-1F-5B : OK
```

However in one of the first launches of the app not a single Asus System Control interface command went through (not only fan curve)

```
12/13 星期五 14:02:28: BatteryLimit = 80 : 0
12/13 星期五 14:02:29: Eco flag : -65536
12/13 星期五 14:02:29: Mux flag : -65536
12/13 星期五 14:02:30: Mode = 2 : 0 <---- nothing is working, even just default mode selection
12/13 星期五 14:02:30: TUF_KB = 983485 : 0
12/13 星期五 14:02:30: VivoMode = 1 : 0
12/13 星期五 14:02:30: AuraMode: AuraColorCycle
12/13 星期五 14:02:30: TUF RGB = B4-02-FF-FF-FF-E1 : 0
12/13 星期五 14:02:30: TUF RGB = B3-02-FF-FF-FF-E1 : 0
12/13 星期五 14:02:30: FanCPU = 1E-38-3D-42-4B-52-58-61-00-01-02-0A-0A-10-21-5A : 0
12/13 星期五 14:02:30: FanGPU = 1E-38-3D-42-4C-52-58-61-00-01-07-0E-1F-1F-1F-5B : 0
```

So most likely it's some Asus driver or firmware malfunction that just solved by itself after reboot. It has nothing to do with app version.

Thanks
